tests/integration/test_applications_send_logs.py

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import asyncio
+import logging
+from pathlib import Path
+
+import pytest
+import yaml
+from helpers import is_loki_up, loki_api_query, oci_image
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+resources = {"loki-image": METADATA["resources"]["loki-image"]["upstream-source"]}
+tester_resources = {
+    "workload-image": oci_image(
+        "./tests/integration/log-proxy-tester/metadata.yaml", "workload-image"
+    )
+}
+
+
+# WARNING: the Loki query API is very finicky about quoting. It MUST be surrounded
+# with single quotes and have double quotes inside, or the return code will be 400.
+#
+# Proceed with care
+tester_apps = {
+    "loki-tester": '{juju_application="loki-tester",level="debug"}',
+    "log-proxy-tester-file": '{juju_application="log-proxy-tester-file",filename=~".+"}',
+    "log-proxy-tester-syslog": '{juju_application="log-proxy-tester-syslog",job=~".+syslog"}',
+}
+
+
+loki_app_name = "loki"
+tester_app_names = list(tester_apps.keys())
+app_names = [loki_app_name] + tester_app_names
+
+
+@pytest.mark.abort_on_fail
+async def test_loki_api_client_logs(
+    ops_test, loki_charm, loki_tester_charm, log_proxy_tester_charm
+):
+    """Test basic functionality of Loki push API relation interface."""
+    await asyncio.gather(
+        ops_test.model.deploy(
+            loki_charm,
+            resources=resources,
+            application_name=loki_app_name,
+        ),
+        ops_test.model.deploy(loki_tester_charm, application_name="loki-tester"),
+        ops_test.model.deploy(
+            log_proxy_tester_charm,
+            resources=tester_resources,
+            application_name="log-proxy-tester-file",
+        ),
+        ops_test.model.deploy(
+            log_proxy_tester_charm,
+            resources=tester_resources,
+            application_name="log-proxy-tester-syslog",
+            config={"syslog": "true", "file_forwarding": "false"},
+        ),
+    )
+    await ops_test.model.wait_for_idle(apps=app_names, status="active")
+
+    await asyncio.gather(
+        *[ops_test.model.add_relation(loki_app_name, t) for t in tester_app_names]
+    )
+    await ops_test.model.wait_for_idle(apps=app_names, status="active")
+
+    for query in tester_apps.values():
+        logs_per_unit = await asyncio.gather(
+            loki_api_query(ops_test, loki_app_name, query, unit_num=0),
```

Comment:
Were `unit_num=1` and `unit_num=2` intentionally removed?
If yes, then I would certainly appreciate a code comment about it.